pages/api/templates.ts

```diff
@@ -20,14 +19,31 @@ import {
 } from "@lib/middleware/jsonIDValidator";
 import { Session } from "next-auth";
 import { BetterOmit } from "@lib/types";
+import { MiddlewareProps } from "@lib/types";
+import { Ability, createAbility } from "@lib/policyBuilder";
 
 const allowedMethods = ["GET", "POST", "PUT", "DELETE"];
 const authenticatedMethods = ["POST", "PUT", "DELETE"];
 
-const templates = async (req: NextApiRequest, res: NextApiResponse) => {
+const templates = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  { session }: MiddlewareProps
+) => {
   try {
-    const session = await isAdmin({ req, res });
-    const response = await templateCRUD({ user: session?.user, method: req.method, ...req.body });
+    if (!session) {
+      res.status(403);
```

Comment:
Let's swap this out to a `401` instead of a `403`.  
`401`  : Unauthorized (do not have a session
`403`  :  Forbidden (not allowed to perform action)